src/components/board.tsx

```diff
@@ -124,81 +130,222 @@ export default class Board extends React.Component<Props, State>{
-        return pieceTrain = this.doMatch(index)
-      } 
+      if (pieceTrain.length >= 3 && !pieceTrain.includes(tile.pieceType)) {
+        return (pieceTrain = this.doMatch(index, "horizontal"));
+      }
 
-      if(tile.pieceType === "[]") return pieceTrain = [];
+      if (tile.pieceType === "[]") return (pieceTrain = []);
 
-      if(!pieceTrain.length) return pieceTrain = [tile.pieceType]
-      console.log("nope", pieceTrain, tile)
-      return pieceTrain = [tile.pieceType];
+      if (!pieceTrain.length) return (pieceTrain = [tile.pieceType]);
+      // console.log("nope", pieceTrain, tile);
+      return (pieceTrain = [tile.pieceType]);
     });
+  }
+
+  searchY() {
+    const { board } = this.state;
+    let pieceTrain: Array<PieceType> = [];
+    for (let x = 0; x < 6; x++) {
+      for (let y = 0; y < board.length; y += 6) {
+        const tile = board[y + x];
+        // console.log(tile, y, pieceTrain);
+        if (y % board.length === 0) {
+          // console.log("------------NEW COLUMN-------------");
+          pieceTrain = board[y + x].pieceType === "[]" ? [] : [tile.pieceType];
+          continue;
+        }
+
+        if (pieceTrain.includes(tile.pieceType)) {
+          pieceTrain = this.addToTrain(
+            pieceTrain,
+            tile.pieceType,
+            y + x,
+            "vertical"
+          );
+          continue;
+        }
+
+        // match 3
+        if (pieceTrain.length >= 3 && !pieceTrain.includes(tile.pieceType)) {
+          pieceTrain = this.doMatch(y + x, "vertical");
+          continue;
+        }
+
+        if (tile.pieceType === "[]") {
+          pieceTrain = [];
+          continue;
+        }
+
+        if (!pieceTrain.length) {
+          pieceTrain = [tile.pieceType];
+          continue;
+        }
+        // console.log("nope", pieceTrain, tile);
+        pieceTrain = [tile.pieceType];
+      }
+    }
+  }
+
+  applyCombo() {
+    // send blocks to other side and/or score multiplier
+    console.log("APPLYING COMBO:", `x${Math.floor(this.state.combo / 3)}`);
+  }
 
-    //TODO: search vertical matches
-    board.forEach((tile) => {
-    //   console.log(tile);
+  searchForMatch() {
+    this.setState({ comboCount: 0 });
+    // search x
+    this.searchX();
+
+    //search y
+    this.searchY();
+
+    this.applyCombo();
+    if (this.state.comboCount) {
+      this.dropPieces(2000)
+    } else {
+      this.setState({combo: 0});
+    }
+  }
+
+  async dropPieces(wait: number) {
+    // 2 second timer before drop
+    console.log("wait for it..");
+    this.setState({ paused: true });
+    await new Promise((resolve) => setTimeout(resolve, wait));
+
+    const { board } = this.state;
+    let newBoard = [...board];
+
+    // traverse vertically to find where blocks should fall
+    for (let x = 0; x < 6; x++) {
+      let column: Array<Tile> = [];
+      for (let y = 0; y < board.length; y += 6) {
+        column = [...column, board[x + y]];
```

Comment:
nice :balloon:

Reply:
should use props columns here (when converting this to a functional component, could get right of the board state for column row player)
```suggestion
    // traverse vertically to find where blocks should fall
    for (let x = 0; x < this.props.column; x++) {
      let column: Array<Tile> = [];
      for (let y = 0; y < board.length; y += this.props.column) {
        column = [...column, board[x + y]];
```

Reply:
yeah, removing magic numbers is a better idea 👍